Library/Control/ControlAccess.hh

```diff
@@ -0,0 +1,105 @@
+/*
+ * ControlAccess.hh
+ *
+ *  Created on: Sept 14, 2019
+ *      Author: N.S. Oblath
+ */
+
+#ifndef NYMPH_CONTROLACCESS_HH_
+#define NYMPH_CONTROLACCESS_HH_
+
+#include "Controller.hh"
+
+#include "Exception.hh"
+#include "MemberVariable.hh"
+#include "ReturnBuffer.hh"
+
+#include "singleton.hh"
+
+namespace Nymph
+{
+
+    class ControlAccess : public scarab::singleton< ControlAccess >, public ControllerInterface
+    {
+        protected:
+            allow_singleton_access( ControlAccess );
+
+            ControlAccess();
+            virtual ~ControlAccess();
+
+        public:
+            virtual bool WaitToContinue();
+
+            /// Returns when processing is completed or a breakpoint is reached
+            /// If the return is true, processing can continue after the break
+            /// If the return is false, processing has ended (either normally or due to an error)
+            virtual bool WaitForBreakOrCanceled();
+
+            /// Use this to have a thread wait for the end of a run
+            virtual void WaitForEndOfRun();
+
+            /// Instruct the Controller to continue after a breakpoint
+            virtual void Continue();
+
+            /// Cancel all threads and end the run
+            virtual void Cancel( int code = 0 );
+
+            /// Reports whether controls is canceled
+            virtual bool IsCanceled() const;
+
+            /// Initiate a break
+            virtual void Break();
+
+            /// Reports whether control is at a breakpoint
+            virtual bool IsAtBreak() const;
+
+            /// Notify the control that a chain is quitting
+            virtual void ChainIsQuitting( const std::string& name, std::exception_ptr ePtr = std::exception_ptr() );
+
+            MEMVAR( ControllerInterface*, Control );
+
+    };
+
+
+
+    class ReturnAccess : public scarab::cancelable
```

Comment:
This could use a comment or two to describe the class?

Reply:
I'm going to address the return capability later.